probe/pkg/probe/probe.go

```
@@ -2,21 +2,270 @@ package probe
+	if err != nil {
+		return nil, errors.Wrap(err, "ensure central state failed")
+	}
+	return centralResp, nil
+}
+
+func (p *ProbeImpl) ensureCentralDeleted(ctx context.Context, centralRequest *public.CentralRequest) error {
+	ensureFunc := func(context.Context) error {
+		_, response, err := p.fleetManagerClient.GetCentralById(ctx, centralRequest.Id)
+		if err != nil {
+			if response != nil && response.StatusCode == http.StatusNotFound {
+				glog.Infof("central instance %s has been deleted", centralRequest.Id)
+				return nil
+			}
+			return errors.Wrapf(err, "central instance %s not reachable", centralRequest.Id)
+		}
+		return errors.Errorf("central instance %s not deleted", centralRequest.Id)
+	}
+
+	if err := retryUntilSucceeded(ctx, ensureFunc, p.config.ProbePollPeriod); err != nil {
+		return errors.Wrap(err, "ensure central deleted failed")
+	}
+	return nil
+}
+
+func (p *ProbeImpl) pingURL(ctx context.Context, url string) error {
+	pingFunc := func(context.Context) error {
+		request, err := http.NewRequestWithContext(ctx, http.MethodHead, url, nil)
+		if err != nil {
+			return errors.Wrap(err, "failed to create request")
+		}
+		response, err := p.httpClient.Do(request)
+		if err != nil {
+			return errors.Wrap(err, "URL not reachable")
+		}
+		defer response.Body.Close()
+		if !httputil.Is2xxStatusCode(response.StatusCode) {
+			return errors.Errorf("URL ping did not succeed: %s", response.Status)
+		}
+		return nil
+	}
+
+	if err := retryUntilSucceeded(ctx, pingFunc, p.config.ProbePollPeriod); err != nil {
+		return errors.Wrap(err, "URL ping failed")
+	}
+	return nil
+}
+
+func retryUntilSucceeded(ctx context.Context, fn func(context.Context) error, interval time.Duration) error {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "retry failed")
+		case <-ticker.C:
+			err := fn(ctx)
+			if err != nil {
+				glog.Warning(err)
```

Should we log the error here or silently ignore it?
Currently, we do log every error in `cleanUpFunc` and the default error we return pretty much doesn't say a lot (`central clean up not successful`).
I know that it's not necessarily given that this function knows the details of the passed `fn` - so either we should stop logging within `cleanUpFunc` and propagate the whole error context and log it here, or silently ignore here and log within `cleanUpFunc`.